pkg/odo/cli/service/list.go

```diff
@@ -39,23 +42,57 @@ func NewServiceListOptions() *ServiceListOptions {
 
 // Complete completes ServiceListOptions after they've been created
 func (o *ServiceListOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.Context = genericclioptions.NewContext(cmd)
+	if experimental.IsExperimentalModeEnabled() {
```

Comment:
Wouldn't it make more sense to put that logic in NewContext so that the same logic doesn't need to be implemented everywhere?

Reply:
Makes sense to me. That's going to change the way bunch of things are happening right now. Specially in the world of devfile. Do you mind if I open a new issue for this?

Reply:
@metacosm I've created #3154 to track your suggestion.